tutorials/haskell/haskell-time-tutorial/tutorial.md

```diff
@@ -0,0 +1,481 @@
+as an argument is in. As a class constrait we have that the period where passing
+must be part of the `Ord` type class.
+
+So after this one was defined all we had to do was create the instances
+of the type class for the different data types: `Year`, `Quarter`, `Month`
+and `Day` itself, so here it’s the code:
+
+```haskell
+instance DayPeriod Year where
+   periodFirstDay y = YearMonthDay y January 1
+   periodLastDay y = YearMonthDay y December 31
+   dayPeriod (YearMonthDay y _ _) = y
+```
+
+Another pattern synonym is introduced here. The `YearMontDay` pattern lets
+us build a `Day` in terms of the `Year`, `MonthOfYear` (do you remember
+this one?) and `DayOfMOnth`, where all three are basically `Int`s. Here is
+its definition:
+
+```haskell
+pattern YearMonthDay :: Year -> MonthOfYear -> DayOfMonth -> Day
+pattern YearMonthDay y m d <-
+   (toGregorian -> (y, m, d))
+   where
+       YearMonthDay y m d = fromGregorian y m d
+```
+
+Now, let’s check the `Quarter` instance:
+
+```haskell
+instance DayPeriod Quarter where
+   periodFirstDay (YearQuarter y q) =
+       case q of
+           Q1 -> periodFirstDay $ YearMonth y January
+           Q2 -> periodFirstDay $ YearMonth y April
+           Q3 -> periodFirstDay $ YearMonth y July
+           Q4 -> periodFirstDay $ YearMonth y October
+   periodLastDay (YearQuarter y q) =
+       case q of
+           Q1 -> periodLastDay $ YearMonth y March
+           Q2 -> periodLastDay $ YearMonth y June
+           Q3 -> periodLastDay $ YearMonth y September
+           Q4 -> periodLastDay $ YearMonth y Dec
+```
+
+Which is defined in terms of the `periodFirstDay` and `periodLastDay` of a `Month`
+and it uses (once again) a pattern named `YearMonth`.
+
+
+Now let’s check the `Month` instance of `DayPeriod`:
+
+```haskell
+instance DayPeriod Month where
+   periodFirstDay (YearMonth y m) = YearMonthDay y m 1
+   periodLastDay (YearMonth y m) = YearMonthDay y m 31 -- clips to correct day
+   dayPeriod (YearMonthDay y my _) = YearMonth y my
+```
+
+Apparently patterns are everywhere, and why not? They are pretty useful.
+
+And last but not least, let's review the `Day` instance of `DayPeriod`:
+
+```haskell
+instance DayPeriod Day where
+   periodFirstDay = id
+   periodLastDay = id
+   dayPeriod = id
+```
+
+As you could imagine, the first and last day of a day is the day itself.
+
+### Some `time` new features examples.
+So let's see what we can do with these new features introduced in the contribution.
+In this section we will review some concrete code samples of how these new features
+can help us in our daily software development routine. After making the proper imports
+we can have something like this:
+
+- Let's compare the first function that I presented, how to get a month's boundaries.
+As we can see this function is more intuitive when comparing with the first one,
+and the functions that we use now are self-decriptivec
+
+```haskell
```

Comment:
What happens if you indent this block so that it's part of the item? I think it should look better.

Reply:
It worked :)